sdcm/utils/adaptive_timeouts/load_info_store.py

```diff
@@ -0,0 +1,211 @@
+    def read_bandwidth_mb(self) -> float:
+        """based on io_properties.yaml in MB/s """
+        return self._io_properties["disks"][0]["read_bandwidth"] / 1024 / 1024
+
+    @cached_property
+    def write_bandwidth_mb(self) -> float:
+        """based on io_properties.yaml in MB/s"""
+        return self._io_properties["disks"][0]["write_bandwidth"] / 1024 / 1024
+
+    @cached_property
+    def read_iops(self) -> float:
+        return self._io_properties["disks"][0]["read_iops"]
+
+    @cached_property
+    def write_iops(self) -> float:
+        return self._io_properties["disks"][0]["write_iops"]
+
+    def as_dict(self):
+        return {
+            "node_name": self._name,
+            "cpu_load_5": self.cpu_load_5,
+            "shards_count": self.shards_count,
+            "read_bandwidth_mb": self.read_bandwidth_mb,
+            "write_bandwidth_mb": self.write_bandwidth_mb,
+            "read_iops": self.read_iops,
+            "write_iops": self.write_iops,
+            "node_data_size_mb": self.node_data_size_mb,
+            "scylla_version": self._scylla_version,
+        }
+
+
+class AdaptiveTimeoutStore(metaclass=Singleton):
+    """Class for storing metrics and other info related to adaptive timeouts.
+
+    Used for future reference/node operations time tracking and calculations optimization."""
+
+    # pylint: disable=too-many-arguments
+    def store(self, metrics: dict[str, Any], operation: str, duration: int | float, timeout: int,
+              timeout_occurred: bool) -> None:
+        pass
+
+    def get(self, operation: str | None, timeout_occurred: bool = False):
+        pass
+
+
+class ESAdaptiveTimeoutStore(AdaptiveTimeoutStore):
+    """AdaptiveTimeoutStore implementation backed by Elasticsearch."""
+
+    def __init__(self):
+        self._es = ES()
+        self._test_id = TestConfig.test_id()
+        self._index = "sct-adaptive-timeouts"
+
+    # pylint: disable=too-many-arguments
+    def store(self, metrics: dict[str, Any], operation: str, duration: float, timeout: float,
+              timeout_occurred: bool):
+        body = metrics
+        body["test_id"] = self._test_id
+        body["operation"] = operation
+        body["duration"] = duration
+        body["timeout"] = timeout
+        body["timeout_occurred"] = timeout_occurred
+        load_id = str(uuid.uuid4())
+        LOGGER.debug("Storing adaptive timeout info: %s=%s", load_id, body)
+        # pylint: disable=unexpected-keyword-arg,no-value-for-parameter
+        self._es.create(index=self._index, id=load_id, document=body)
+
+    def get(self, operation: str | None, timeout_occurred: bool | None = None):
+        """Get adaptive timeout info from ES.
+
+        Example usage:
+            >>> from sdcm.utils.adaptive_timeouts.load_info_store import ESAdaptiveTimeoutStore
+            >>> from sdcm.utils.adaptive_timeouts import Operations
+            >>> ESAdaptiveTimeoutStore().get(operation=Operations.DECOMMISSION.name)
```

Comment:
I updated some code to view metrics/data from ES with example. I already fixed missing version details. 

Also looks like this calculation needs some tuning - when CPU usage is high it soft-times-out